src/campaigns/tasks/services/aiCampaignManager.service.ts

```diff
@@ -0,0 +1,214 @@
+    this.apiBaseUrl = process.env.AI_CAMPAIGN_MANAGER_BASE
+  }
+
+  async startCampaignPlanGeneration(
+    request: StartCampaignPlanRequest,
+  ): Promise<CampaignPlanSession> {
+    try {
+      const response = await this.fetchFromApi<CampaignPlanSession>(
+        '/start-campaign-plan-generation',
+        {
+          method: Methods.POST,
+          data: request,
+        },
+      )
+
+      return response.data
+    } catch (error) {
+      this.logger.error('Failed to start campaign plan generation', error)
+      throw new BadGatewayException('Failed to start campaign plan generation')
+    }
+  }
+
+  async downloadJson(sessionId: string): Promise<unknown> {
+    try {
+      const response = await this.fetchFromApi(`/download-json/${sessionId}`, {
+        method: Methods.GET,
+      })
+
+      return response.data
+    } catch (error) {
+      this.logger.error('Failed to download campaign plan JSON', error)
+      throw new BadGatewayException('Failed to download campaign plan JSON')
+    }
+  }
+
+  async getProgressStream(sessionId: string): Promise<ProgressStreamData[]> {
+    try {
+      const response = await this.fetchFromApi<string>(
+        `/progress-stream/${sessionId}`,
+        {
+          method: Methods.GET,
+        },
+      )
+
+      const lines = response.data
+        .split('\n')
+        .filter((line) => line.startsWith('data: '))
+      const progressData: ProgressStreamData[] = []
+
+      for (const line of lines) {
+        try {
+          const jsonStr = line.replace('data: ', '')
+          const data = JSON.parse(jsonStr) as ProgressStreamData
+          progressData.push(data)
+        } catch (parseError) {
+          this.logger.warn('Failed to parse progress line', {
+            line,
+            parseError,
+          })
+        }
+      }
+
+      return progressData
+    } catch (error) {
+      this.logger.error('Failed to get progress stream', error)
+      throw new BadGatewayException('Failed to get progress stream')
+    }
+  }
+
+  async waitForCompletion(
+    sessionId: string,
+    maxWaitTimeMs = 300000, // 5 minutes default
+    pollIntervalMs = 5000, // 5 seconds default
+  ): Promise<ProgressStreamData> {
+    const startTime = Date.now()
+
+    while (Date.now() - startTime < maxWaitTimeMs) {
+      const progressData = await this.getProgressStream(sessionId)
+      const latestProgress = progressData[progressData.length - 1]
+
+      if (latestProgress?.status === 'completed') {
+        return latestProgress
+      }
+
+      if (latestProgress?.status === 'failed') {
+        throw new BadGatewayException('Campaign plan generation failed')
+      }
+
+      await this.sleep(pollIntervalMs)
+    }
+
+    throw new BadGatewayException('Campaign plan generation timed out')
+  }
+
+  private async fetchFromApi<T>(
+    endpoint: string,
+    options: {
+      method?: Methods
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      data?: any
```

Comment:
This is why I don't abstract out simple fetch calls in services.. if you want to do this, please don't override `any` to cause confusion for others who will have to inevitably work on this.  Please use a generic and type it across it's usages.